### PR TITLE
Support Getting Outputs in multiple formats

### PR DESCRIPTION
**Pull request by suhasgumma**
## Changes Made:
* Added support for getting outputs in multiple formats.
* Currently, the required formats are passed in a `single string separated by commas(,)` to the formats flag. Could change the input of formats flag to slice if required in the future.

* Takes a single output file as input and adds the required extensions to generate multiple output files for multiple formats.

Multi Format

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/pkg/resultshandling/results.go [hunk 1]
```diff
@@ -65,13 +65,19 @@ func (resultsHandler *ResultsHandler) GetResults() *reporthandlingv2.PostureRepo
...
+	for _, printerObj := range printerObjs {
+		printerObj.ActionPrint(resultsHandler.scanData)
+	}
 
 	if err := resultsHandler.reporterObj.Submit(resultsHandler.scanData); err != nil {
 		return err
 	}
 
-	resultsHandler.printerObj.Score(resultsHandler.GetRiskScore())
+	for _, printerObj := range printerObjs {
```
Comment (dwertent) [inline core/pkg/resultshandling/results.go:78]:
```suggestion
	for i := range printerObjs {
```

File: core/pkg/resultshandling/results.go [hunk 2]
```diff
@@ -65,13 +65,19 @@ func (resultsHandler *ResultsHandler) GetResults() *reporthandlingv2.PostureRepo
...
+		printerObj.Score(resultsHandler.GetRiskScore())
```
Comment (dwertent) [inline core/pkg/resultshandling/results.go:79]:
```suggestion
		printerObjs[i].Score(resultsHandler.GetRiskScore())
```

File: core/pkg/resultshandling/results.go [hunk 3]
```diff
@@ -65,13 +65,19 @@ func (resultsHandler *ResultsHandler) GetResults() *reporthandlingv2.PostureRepo
 // HandleResults handle the scan results according to the pre defined interfaces
 func (resultsHandler *ResultsHandler) HandleResults() error {
 
-	resultsHandler.printerObj.ActionPrint(resultsHandler.scanData)
+	printerObjs := resultsHandler.printerObjs
+
+	for _, printerObj := range printerObjs {
```
Comment (dwertent) [inline core/pkg/resultshandling/results.go:70]:
```suggestion
	for i := range printerObjs {
		printerObjs[i].ActionPrint(resultsHandler.scanData)
	}
```
I dont think we should iterate over the objects.

File: core/pkg/resultshandling/printer/v2/sarifprinter.go [hunk 4]
```diff
@@ -144,6 +144,8 @@ func (sp *SARIFPrinter) ActionPrint(opaSessionObj *cautils.OPASessionObj) {
 	report.AddRun(run)
 
 	report.PrettyWrite(sp.writer)
+
+	printer.LogOutputFile(sp.writer.Name())
```
Comment (dwertent) [inline core/pkg/resultshandling/printer/v2/sarifprinter.go:148]: Well done.

File: core/pkg/resultshandling/printer/v1/jsonprinter.go [hunk 5]
```diff
@@ -42,4 +42,6 @@ func (jsonPrinter *JsonPrinter) ActionPrint(opaSessionObj *cautils.OPASessionObj
 		logger.L().Fatal("failed to convert posture report object")
 	}
 	jsonPrinter.writer.Write(postureReportStr)
```
Comment (dwertent) [inline core/pkg/resultshandling/printer/v1/jsonprinter.go:44]: We should handle the error here the same way we handled in other places

File: core/cautils/scaninfo.go [hunk 6]
```diff
@@ -188,25 +187,39 @@ func (scanInfo *ScanInfo) setUseFrom() {
...
 }
 
-func (scanInfo *ScanInfo) setOutputFile() {
-	if scanInfo.Output == "" {
-		return
-	}
-	if scanInfo.Format == "json" {
-		if filepath.Ext(scanInfo.Output) != ".json" {
-			scanInfo.Output += ".json"
+func (scanInfo *ScanInfo) GetFormats() []string {
```
Comment (dwertent) [inline core/cautils/scaninfo.go:190]:
```suggestion
func (scanInfo *ScanInfo) getFormats() []string {
```
Is there a reason for this function to be public?

File: core/cautils/scaninfo.go [hunk 7]
```diff
@@ -188,25 +187,39 @@ func (scanInfo *ScanInfo) setUseFrom() {
...
+	return strings.Split(scanInfo.Format, ",")
+}
+
+func (scanInfo *ScanInfo) GetOutputFiles() []string {
+	formats := scanInfo.GetFormats()
+	outputs := make([]string, 0)
+
+	for _, format := range formats {
+		if scanInfo.Output == "" {
```
Comment (dwertent) [inline core/cautils/scaninfo.go:199]: This should be before iterating over `formats`

File: core/cautils/scaninfo.go [hunk 8]
```diff
@@ -188,25 +187,39 @@ func (scanInfo *ScanInfo) setUseFrom() {
...
+			outputs = append(outputs, "")
```
Comment (dwertent) [inline core/cautils/scaninfo.go:200]: The default should be `pretty-printer`

File: core/cautils/scaninfo.go [hunk 9]
```diff
@@ -188,25 +187,39 @@ func (scanInfo *ScanInfo) setUseFrom() {
...
 		}
-	}
-	if scanInfo.Format == "junit" {
-		if filepath.Ext(scanInfo.Output) != ".xml" {
-			scanInfo.Output += ".xml"
+
+		output := scanInfo.Output
+
+		if format == "json" {
```
Comment (dwertent) [inline core/cautils/scaninfo.go:205]: The interface should include the "printer name" and "printer extension" functions, and we should have an object that lists the supported outputs, this way we can iterate based on name and extension so we do not need this `if` situation.

**Review by dwertent — CHANGES_REQUESTED** [review]

File: core/cautils/scaninfo.go (hunk 6 above)
Comment (suhasgumma) [reply core/cautils/scaninfo.go:190]: `GetFormats` is defined in `cautils` package and called in both `cautils` and `core` packages.

File: core/pkg/resultshandling/printer/v1/jsonprinter.go (hunk 5 above)
Comment (suhasgumma) [reply core/pkg/resultshandling/printer/v1/jsonprinter.go]: Done!!

File: core/pkg/resultshandling/results.go (hunk 3 above)
Comment (suhasgumma) [reply core/pkg/resultshandling/results.go]: Done. Made the suggested changes.

File: core/pkg/resultshandling/results.go (hunk 1 above)
Comment (suhasgumma) [reply core/pkg/resultshandling/results.go]: Done.

File: core/pkg/resultshandling/results.go (hunk 2 above)
Comment (suhasgumma) [reply core/pkg/resultshandling/results.go]: Done.

**Comment by suhasgumma** [comment]
Sorry, @dwertent !! Made unnecessary merge. I will close and open a new PR.

**Pull request closed without merging**